src/server/server.ts

```diff
@@ -210,6 +212,40 @@ registerRoute('post', '/compile/snippet', compileCommon);
  */
 registerRoute('post', '/compile/page', (req, res) => compileCommon(req, res, true /*wrapWithRunnerChrome*/));
 
+registerRoute('get', '/open-in-playground/:host/:type/:id/:filename', async (req, res) => {
+    let relativePath;
+    switch (req.params.host.toUpperCase()) {
+        case 'EXCEL':
+            relativePath = 'xl';
+            break;
+        case 'WORD':
+            relativePath = 'word';
+            break;
+        case 'POWERPOINT':
+            relativePath = 'ppt';
+            break;
+        default:
+            return;
+    }
+
+    return Promise.resolve(getVersionNumber()).then(versionNumber => {
+        let extractDirName = path.resolve(__dirname, `test${(new Date()).getTime()}`);
+        let zip = fs.createReadStream(path.resolve(__dirname, 'Doc1.zip')).pipe(unzip.Extract({ path: extractDirName }));
```

Comment:
Need to generate random file name, if two requests come in at the same time there will be a conflict with the filename and user2 could get data from user1.